src/librustc_lint/types.rs

```diff
@@ -176,6 +177,23 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TypeLimits {
                             _ => bug!(),
                         };
                         if lit_val < min || lit_val > max {
+                            let parent_id = cx.tcx.hir.get_parent_node(e.id);
+                            if let hir_map::NodeExpr(parent_expr) = cx.tcx.hir.get(parent_id) {
+                                if let hir::ExprCast(..) = parent_expr.node {
+                                    if let ty::TyChar = cx.tables.expr_ty(parent_expr).sty {
+                                        let mut err = cx.struct_span_lint(
+                                                             OVERFLOWING_LITERALS,
+                                                             e.span,
+                                                             "only u8 can be casted into char");
+                                        err.span_suggestion(e.span,
+                                                            &format!("try with `'\\u{{{:X}}}'`",
+                                                                     lit_val),
+                                                            "".to_owned());
```

Comment:
This should be `format!("'\\u{{{:X}}}'", lit_val)`. Otherwise when tools try to apply this will remove the code from the users file.